tests/e2e/taqueria-plugin-flextesa.spec.ts

```diff
@@ -235,4 +237,236 @@ describe("E2E Testing for taqueria flextesa plugin",  () => {
         }
     })
 
-});
\ No newline at end of file
+});
+
+// TODO: to be moved into a different spec file when the tezos-client plugin no longer depends on the flextesa plugin.
+// If I move it now, testing won't work.
+// Starting docker in e2e/auto-test-flextesa-plugin and then e2e/auto-test-tezos-client-plugin won't work
+// because Jest parallelizes test runs across spec files it seems.
+// The volumes end up being different, and so the tests below will complain that files don't exist.
+describe("E2E Testing for taqueria typechecker and simulator tasks of the tezos-client plugin", () => {
```

Comment:
Due to Jest parallelization (more detail in the code comment above the test suite), I am leaving it in flextesa.spec for now. It would be easy to move it to a new spec file in the future; just need to cut the entire `describe()` function. But I don't have the expertise make it work today.

Reply:
I have a solution that doesn't require some sort of special Jest knowledge. But that requires a quite significant amount of changes.

The solution is to have an extra parameter for the container name (`docker run --name <the name>`) when starting one. Right now it's "hardcoded" (in a sense) to be the same as the sandbox name.

Reply:
I like that approach @jchenche 👍

On a related, but different note:

Right now there is an issue where we can only launch a single sandbox at a time (even when different ports are specified:
```
❯ taq start sandbox local
Error: Command failed: sh -c "docker run --name local --rm --detach --platform linux/arm64/v8 -p 20000:20000 -p 19229:9229 -v /Users/houstonkrohman/code/taqueria-scaffold-demo:/project -w /app ghcr.io/ecadlabs/taqueria-flextesa:dev-557-typechecker-task-JCC-200406f node index.js --sandbox local"
docker: Error response from daemon: driver failed programming external connectivity on endpoint local (2dc5d41c650b90ac7b05d80557407a395bd0b0a2da35da9ffdfb96adb3d034c2): Bind for 0.0.0.0:19229 failed: port is already allocated.

    at ChildProcess.exithandler (node:child_process:399:12)
    at ChildProcess.emit (node:events:520:28)
    at maybeClose (node:internal/child_process:1090:16)
    at Socket.<anonymous> (node:internal/child_process:449:11)
    at Socket.emit (node:events:520:28)
    at Pipe.<anonymous> (node:net:687:12) {
  killed: false,
  code: 125,
  signal: null,
  cmd: 'sh -c "docker run --name local --rm --detach --platform linux/arm64/v8 -p 20000:20000 -p 19229:9229 -v /Users/houstonkrohman/code/taqueria-scaffold-demo:/project -w /app ghcr.io/ecadlabs/taqueria-flextesa:dev-557-typechecker-task-JCC-200406f node index.js --sandbox local"'
}
```
first sandbox is running on port 2020, tried starting local on port 2000...

Is this hardcoding related to that?

Reply:
hmm not sure about this one.